addons/hr_recruitment_skills/models/hr_applicant.py

```diff
@@ -32,3 +32,15 @@ def create_employee_from_applicant(self):
             'skill_type_id': applicant_skill.skill_type_id.id,
         }) for applicant_skill in self.applicant_skill_ids]
         return action
+
+    def update_employee_from_applicant(self):
+        for applicant in self:
+            existing_skills = applicant.emp_id.employee_skill_ids.skill_id
+            skills_to_create = applicant.applicant_skill_ids.skill_id - existing_skills
+            applicant.emp_id.employee_skill_ids |= self.env['hr.employee.skill'].create([{
+                'employee_id': applicant.emp_id.id,
+                'skill_id': skill.id,
+                'skill_level_id': applicant.applicant_skill_ids.filtered(lambda s: s.skill_id == skill).skill_level_id.id,
+                'skill_type_id': skill.skill_type_id.id,
+            } for skill in skills_to_create])
+        return super().update_employee_from_applicant()
```

Comment:
You don't need to assign the newly created values to `employee_skill_ids`, hence you can group all the vals together and call create() only once, something like:

```suggestion
    def update_employee_from_applicant(self):
        create_vals = []
        for applicant in self:
            existing_skills = applicant.emp_id.employee_skill_ids.skill_id
            skills_to_create = applicant.applicant_skill_ids.skill_id - existing_skills
            create_vals.extend([{
                'employee_id': applicant.emp_id.id,
                'skill_id': skill.id,
                'skill_level_id': applicant.applicant_skill_ids.filtered(lambda s: s.skill_id == skill).skill_level_id.id,
                'skill_type_id': skill.skill_type_id.id,
            } for skill in skills_to_create])
        self.env['hr.employee.skill'].create(create_vals)
        return super().update_employee_from_applicant()
```